tfs/src/main/java/hudson/plugins/tfs/rm/ConnectReleaseWebHookEvent.java

```diff
@@ -0,0 +1,319 @@
+        final Object resource = event.getResource();
+        final ReleaseWebHookResource parameters = mapper.convertValue(resource, ReleaseWebHookResource.class);
+
+        if (event.getEventType().equalsIgnoreCase(CREATE_WEBHOOK_EVENTNAME)) {
+            createReleaseWebHook(parameters);
+        } else if (event.getEventType().equalsIgnoreCase(REMOVE_WEBHOOK_ENENTNAME)) {
+            deleteReleaseWebHook(parameters);
+        } else if (event.getEventType().equalsIgnoreCase(LIST_WEBHOOK_EVENTNAME)) {
+            return listReleaseWebHook();
+        } else if (event.getEventType().equalsIgnoreCase(LINK_WEBHOOK_EVENTNAME)) {
+            linkWebHook(parameters);
+        } else if (event.getEventType().equalsIgnoreCase(UNLINK_WEBHOOK_EVENTNAME)) {
+            unlinkWebHook(parameters);
+        } else {
+            throw new UnsupportedOperationException("Webhook operation " + parameters.getOperationType() + " is not supported");
+        }
+
+        return JSONObject.fromObject(event);
+    }
+
+    private String validateAndGetPayloadUrl(final ReleaseWebHookResource parameters) {
+        String payloadUrl = parameters.getPayloadUrl();
+
+        if (StringUtils.isBlank(payloadUrl)) {
+            throw new InvalidParameterException("pyaloadUrl is empty");
+        }
+
+        final URI uri;
+        try {
+            uri = new URI(payloadUrl);
+        } catch (final URISyntaxException e) {
+            throw new InvalidParameterException("Malformed Payload URL " + e.getMessage());
+        }
+
+        final String hostName = uri.getHost();
+        if (StringUtils.isBlank(hostName)) {
+            throw new InvalidParameterException("Malformed Payload URL");
+        }
+
+        return StringUtils.stripEnd(StringUtils.trim(payloadUrl), "/");
+    }
+
+    private AbstractProject validateAndGetJenkinsProject(final ReleaseWebHookResource resource) {
+        if (StringUtils.isEmpty(resource.getProjectName())) {
+            throw new InvalidParameterException("Project name is empty");
+        }
+
+        for (final Item project : Jenkins.getActiveInstance().getAllItems()) {
+            if (project instanceof AbstractProject && project.getName().equalsIgnoreCase(resource.getProjectName())) {
+                return (AbstractProject) project;
+            }
+        }
+
+        throw new InvalidParameterException("Cannot find Jenkins Job with the name " + resource.getProjectName());
+    }
+
+    private ReleaseWebHook validateAndGetReleaseWebHookByName(final String webHookName) {
+        ReleaseWebHook webHook = null;
+        for (ReleaseWebHook webHookItem : ReleaseWebHookHelper.getReleaseWebHookConfigurations()) {
+            if (webHookItem.getWebHookName().equalsIgnoreCase(webHookName)) {
+                webHook = webHookItem;
+                logger.fine(String.format("WebHook found for webhook name %s", webHookName));
```

Comment:
there can be multiple matches as names can be duplicate. picking up the first is fine? isn't it better to keep the name unique?

Reply:
fixed.